### PR TITLE
= str #19803 fix 1 element buffer handling

### PR DESCRIPTION
**Pull request by ALPHA-60**
- Sink.queue and Sink.actorRefWithAck can now handle one element input
  buffers
- Sink.queue now buffers as much upstream elements as declared in its
  InputBuffer attribute, instead of one less.
- Tests for streaming with full buffer now do not depend on default
  input buffer size (16); instead, they set sink input buffer size
  explicitly.

### CODE REVIEW

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ALPHA-60** [comment]
Refs #19803

**Comment by ALPHA-60** [comment]
This PR is related to #19676 and #19654.

File: akka-stream-tests/src/test/scala/akka/stream/scaladsl/ActorRefBackpressureSinkSpec.scala [hunk 1]
```diff
@@ -116,21 +116,49 @@ class ActorRefBackpressureSinkSpec extends AkkaSpec with ScalaFutures with Conve
     }
 
     "keep on sending even after the buffer has been full" in assertAllStagesStopped {
+      val bufferSize = 16
+      val streamElementCount = bufferSize + 4
```
Comment (drewhk) [inline akka-stream-tests/src/test/scala/akka/stream/scaladsl/ActorRefBackpressureSinkSpec.scala]: thanks for splitting it up

**Comment by drewhk** [comment]
LGTM, thanks for fixing this!

**Comment by rkuhn** [comment]
OK TO TEST

**Comment by rkuhn** [comment]
Testing this locally now, will merge if successful.

**Comment by rkuhn** [comment]
merged manually. Thanks a lot @ALPHA-60 !

**Comment by ALPHA-60** [comment]
Thanks!

**Pull request closed without merging**